code/_core/obj/item/weapon/ranged/unique/flamethrower.dm

```diff
@@ -0,0 +1,87 @@
+/obj/item/weapon/ranged/unique/flamethrower/
+	name = "flamethrower"
+	desc = "Mphh Mphh!"
+	desc_extended = "A dated flamethrower. Disarm + Fire to release a blast of compressed Fuel"
+	icon = 'icons/obj/item/weapons/ranged/flamethrower.dmi'
+
+	automatic = TRUE
+
+	allow_reagent_transfer_from = TRUE
+	allow_reagent_transfer_to = TRUE
+
+	shoot_delay = 3
+
+	size = SIZE_5
+
+	shoot_sounds = list('sound/weapons/magic/fractal.ogg')
+
+	projectile = /obj/projectile/bullet/flamer
+	projectile_speed = TILE_SIZE*0.3
+
+	reagents = /reagent_container/weapon/flamethrower
+	var/fuel_per_shot = 2
+	var/fuel_per_airblast = 10
+	uses_until_condition_fall = 100
+	use_loyalty_tag = TRUE
+	use_iff_tag = FALSE
+
+	can_wield = TRUE
+	wield_only = TRUE
+
+	inaccuracy_modifier = 0.2
+
+/obj/item/weapon/ranged/unique/flamethrower/get_ammo_count()
+	return FLOOR((reagents.volume_current/fuel_per_shot),1)
+
+/obj/item/weapon/ranged/unique/flamethrower/can_gun_shoot(mob/caller, atom/object, location, params, check_time, messages)
+	if(get_ammo_count() <= 0)
+		handle_empty(caller)
+		return FALSE
+	return ..()
+
+/obj/item/weapon/ranged/unique/flamethrower/click_on_object(mob/caller, atom/object, location, control, params)
```

Comment:
I'd put this in shoot() instead of click_on_object.

Reply:
Yea. Point.

Reply:
Turns out I actually had it there because otherwise you attempt to spill contents instead of airblasting.